receiver/pluginreceiver/supplied_plugins_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginreceiver
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const pluginDirPath = "../../plugins"
+
+// TestValidateSuppliedPlugins ensures each plugin that ships with the collector loads with the current
+// version of the receiver
+func TestValidateSuppliedPlugins(t *testing.T) {
+	entries, err := os.ReadDir(pluginDirPath)
+	require.NoError(t, err)
+
+	for _, entry := range entries {
+		t.Run(fmt.Sprintf("Loading %s", entry.Name()), func(t *testing.T) {
+			t.Parallel()
+			fullFilePath, err := filepath.Abs(filepath.Join(pluginDirPath, entry.Name()))
+			assert.NoError(t, err, "Failed to determine path of file %s", entry.Name())
+
+			// Load the plugin
+			plugin, err := LoadPlugin(fullFilePath)
+			assert.NoError(t, err, "Failed to load file %s", entry.Name())
+
+			_, err = plugin.RenderComponents(map[string]interface{}{})
```

Comment:
Will this work for every plugin?

It's a bridge we can cross when we get to it if not, but are we trying to make all plugins have defaults for all fields?

Reply:
I believe it should as it doesn't actually load anything from factories but rather applies the loaded config to structs and applies defaults.

I'm mostly just testing does this thing load without issues.

Reply:
Yeah, but if a field isn't populated this will fail, right? Like, if one of the options doesn't have a default? Doesn't really matter for this PR, actually, it's fine as is.

Reply:
Possibly, I will cross that bridge when we come to it. I'm not sure to be honest if that's a case we have.